azurelinuxagent/common/protocol/extensions_config_retriever.py

```diff
@@ -0,0 +1,405 @@
+        if svd_seqNo is not None:
+            svd_seqNo_file_path = os.path.join(conf.get_lib_dir(), _SVD_SEQNO_FILE_NAME)
+            self._save_cache(svd_seqNo_file_path, ustr(svd_seqNo))
+
+    def _remove_cache(self, file_name):
+        try:
+            path = os.path.join(conf.get_lib_dir(), file_name)
+            if os.path.exists(path):
+                os.remove(path)
+        except IOError as e:
+            fileutil.clean_ioerror(e, paths=path)
+            raise ProtocolError("Failed to remove cache: {0}".format(e))
+
+    def _set_saved_vm_id(self, cached_vm_id):
+        path = os.path.join(conf.get_lib_dir(), _VM_ID_FILE_NAME)
+        self._save_cache(path, cached_vm_id)
+
+    def _save_cache(self, local_file, data):
+        try:
+            fileutil.write_file(local_file, data)
+        except IOError as e:
+            fileutil.clean_ioerror(e, paths=[local_file])
+            raise ProtocolError("Failed to write cache: {0}".format(e))
+
+    def _get_saved_sequence_number(self):
+        path = os.path.join(conf.get_lib_dir(), _SEQUENCE_NUMBER_FILE_NAME)
+        if os.path.exists(path):
+            sequence_number = fileutil.read_file(path)
+            if sequence_number is not None:
+                return int(sequence_number)
+        return -1
+
+    def _get_saved_svd_seqNo(self):
+        path = os.path.join(conf.get_lib_dir(), _SVD_SEQNO_FILE_NAME)
+        if os.path.exists(path):
+            svd_seqno = fileutil.read_file(path)
+            if svd_seqno is not None:
+                return int(svd_seqno)
+        return -1
+
+    def _get_saved_incarnation(self):
+        path = os.path.join(conf.get_lib_dir(), _INCARNATION_FILE_NAME)
+        if os.path.exists(path):
+            incarnation = fileutil.read_file(path)
+            if incarnation is not None:
+                return str(incarnation)
+        return -1
+
+    def _get_saved_mode(self):
+        path = os.path.join(conf.get_lib_dir(), _GOAL_STATE_SOURCE_FILE_NAME)
+        if os.path.exists(path):
+            goal_state_source = fileutil.read_file(path)
+            return goal_state_source
+        else:
+            return GOAL_STATE_SOURCE_FABRIC
+
+    def _get_saved_vm_id(self):
+        cached_vm_id = None
+        path = os.path.join(conf.get_lib_dir(), _VM_ID_FILE_NAME)
+        if os.path.exists(path):
+            cached_vm_id = fileutil.read_file(path)
+        return cached_vm_id
+
+    def get_description(self):
+        return "{0} Incarnation={1} SeqNo={2} Reason={3}".format(
+            self._last_mode, self._last_fabric_incarnation, self._last_fast_track_seq_no, self._reason)
+
+    def get_ext_config_file_name(self):
+        if self._last_mode == GOAL_STATE_SOURCE_FASTTRACK:
+            return _EXT_CONF_FILE_NAME.format(_EXT_CONFIG_FAST_TRACK, self._get_last_sequence_number())
+        else:
+            return _EXT_CONF_FILE_NAME.format(_EXT_CONFIG_FABRIC, self._get_last_incarnation())
+
+    def _get_vm_id(self):
```

Comment:
@larohra came across this function: https://github.com/Azure/WALinuxAgent/blob/develop/azurelinuxagent/common/osutil/default.py#L307

it turns out that we already have code to get the vm id (which also checks /sys/class/dmi/id/product_uuid)

Reply:
This code has been removed as we no longer keep state